File: src/main/kotlin/com/elissamc/commands/FriendCommand.java
package com.elissamc.commands;

import cn.nukkit.Player;
import cn.nukkit.Server;
import cn.nukkit.command.Command;
import cn.nukkit.command.CommandSender;
import com.elissamc.core.StorageEngine;

public class FriendCommand extends Command {

    public FriendCommand(String name) {
        super(name);
    }

    @Override
    public boolean execute(final CommandSender sender, final String alias, final String[] args) {
        if (sender instanceof Player) {
            if (sender.isOp()) {
                this.processOpCommand((Player) sender, args);
            } else {
                this.processCommand((Player) sender, args);
            }
        } else {
            sender.sendMessage("This command is only avaiable for players!");
        }
        return true;
    }

    private void processCommand(final Player sender, final String[] args) {
        if (args.length >= 1) {
            String playerName = args[0];
            boolean success = false;
            for (Player p : Server.getInstance().getOnlinePlayers().values()) {
                if (p.getName().equalsIgnoreCase(playerName)) {
                    StorageEngine.getProfile(sender.getUniqueId()).addFriend(
                            p.getUniqueId());
                    sender.sendMessage("Player '" + p.getName()
                            + "' has been ADDED to your friends!");
                    p.sendMessage("Player '" + sender.getName()
                            + "' added you to his/her friends! Add him too! /friend "
                            + sender.getName());

                    success = true;
                }
            }
            if (!success)
                sender.sendMessage("Player not found! Player must be online!");
        } else {
            sender.sendMessage("You must provide player name!");
        }
    }

    private void processOpCommand(final Player sender, final String[] args) {
        //No op commands.
        this.processCommand(sender, args);
    }
}

File: src/main/kotlin/com/elissamc/commands/FriendCommand.kt
package com.elissamc.commands

import cn.nukkit.Player
import cn.nukkit.Server
import cn.nukkit.command.Command
import cn.nukkit.command.CommandSender
import com.elissamc.chat.ChatFormat
import com.elissamc.core.StorageEngine

class FriendCommand(name: String) : Command(name) {

    override fun execute(sender: CommandSender, alias: String, args: Array<String>): Boolean {
        if (sender is Player) {
            if (sender.isOp()) {
                this.processOpCommand(sender, args)
            } else {
                this.processCommand(sender, args)
            }
        } else {
            sender.sendMessage(ChatFormat.error("This command is only available for players!"))
        }
        return true
    }

    private fun processCommand(sender: Player, args: Array<String>) {
        if (args.isNotEmpty()) {
            val playerName = args[0]
            var success = false
            for (p in Server.getInstance().onlinePlayers.values) {
                if (p.name.equals(playerName, ignoreCase = true)) {
                    StorageEngine.getProfile(sender.uniqueId).addFriend(
                            p.uniqueId)
                    sender.sendMessage(ChatFormat.success("Player '" + p.name
                            + "' has been ADDED to your friends!"))
                    p.sendMessage(ChatFormat.success("Player '" + sender.name
                            + "' added you to his/her friends! Add him too! /friend "
                            + sender.name))

                    success = true
                }
            }
            if (!success)
                sender.sendMessage(ChatFormat.error("Player not found! Player must be online!"))
        } else {
            sender.sendMessage(ChatFormat.error("You must provide player name!"))
        }
    }

    private fun processOpCommand(sender: Player, args: Array<String>) {
        //No op commands.
        this.processCommand(sender, args)
    }
}